### DISCUSSION TITLE
Max concurrent threads by job

### DISCUSSION THREAD

**Discussion by justinpeck** [quartznet/quartznet#944] [General] [answered]
Is there a way to limit the number of concurrent instances of a specific job? 

For example, if have a dozen jobs of varying duration and 25 threads in the pool, is there a way to enable concurrent execution for one of the jobs, but limit it to say five of those threads?

**Comment by lahma** [comment, marked as answer]
Unfortunately no. It's either one instance (DisallowConcurrentExecution) or unlimited. You can however limit it yourself for example in Execute by having a Semaphore(Slimi) if you are running one instance, otherwise a distributed lock solution is needed.

**Comment by Sicos1977** [comment]
What you could do is access the scheduler throught the context and count the running jobs.

**Reply by lahma** [reply]
And that would be a snapshot of state. However perfectly fine in non-clustered setup. It gets complex depending on the setup.

**Reply by Sicos1977** [reply]
I agree on that, it will be difficult if you are running multiple instances
